plugins/module_utils/core.py

```diff
@@ -357,3 +357,27 @@ def get_boto3_client_method_parameters(client, method_name, required=False):
     else:
         parameters = list(input_shape.members.keys())
     return parameters
+
+
+def scrub_none_parameters(parameters):
+    """
+    Iterate over a dictionary removing any keys that have a None value
+
+    Reference: https://github.com/ansible-collections/community.aws/issues/251
+    Credit: https://medium.com/better-programming/how-to-remove-null-none-values-from-a-dictionary-in-python-1bedf1aab5e4
+
+    :param parameters: parameter dict
+    :return: parameter dict with all keys = None removed
+    """
+
+    clean_parameters = {}
+
+    for k, v in parameters.items():
+        if isinstance(v, dict):
```

Comment:
See line 380.

Do you think we should do something different?

Reply:
Ah. I misread it.

Reply:
Also, similar to empty sub-dictionaries, should empty lists be removed? (I don't know whether it's a good idea. That's something that should be checked first by someone who knows the APIs better.)

Maybe the two behaviors need to be configurable.

Reply:
The main use case I see is purging parameters that are ``None`` from the args as parse by Ansible(AWS)Module.  

The worst offenders being: ``(type=list, options=...)`` and ``(type=dict, options=...)``.  You've covered the second option, but the first is currently left as an exercise for the reader.  What might be nice is to support optionally descending into lists, dropping None, and None from dicts inside the list...

Reply:
Thanks for the feedback.

@tremble I agree with your main use case.  I have altered the code to only discard params set to None.  Empty dicts or lists will be kept because they are likely wanted.  I know modules for example that will take an empty dict for the `tags` param to remove all tags.

The code will also drop lists set as None so will handle (type=list, options=...).